security/pkg/caclient/keycertbundlerotator.go

```diff
@@ -52,13 +54,16 @@ func NewKeyCertBundleRotator(keycert pkiutil.KeyCertBundle, certUtil util.CertUt
 }
 
 // Start periodically rotates the KeyCertBundle by interacting with the upstream CA.
-// It is a blocking function that should run as a go routine. Not thread safe.
+// It is a blocking function that should run as a go routine. Thread safe.
 func (c *keyCertBundleRotatorImpl) Start(errCh chan<- error) {
+	c.stoppedMutex.Lock()
```

Comment:
In general, using shared memory for communication is not recommended practice in go. 
https://blog.golang.org/share-memory-by-communicating

It's ok to checkin to unblock but please consider rearchitecture the async logic to use channels.

Reply:
OK. I will restructure this later.